staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/validation.go

```diff
@@ -378,3 +380,27 @@ func MapIsCorrelatable(mapType *string) bool {
 	// if a third map type is introduced, assume it's not correlatable. granular is the default if unspecified.
 	return mapType == nil || *mapType == "granular" || *mapType == "atomic"
 }
+
+func hasXValidations(s *schema.Structural) bool {
+	if s == nil {
+		return false
+	}
+	if len(s.XValidations) > 0 {
+		return true
+	}
+	if hasXValidations(s.Items) {
+		return true
+	}
+	if s.AdditionalProperties != nil && hasXValidations(s.AdditionalProperties.Structural) {
+		return true
+	}
+	if s.Properties != nil {
+		for propName := range s.Properties {
+			prop := s.Properties[propName]
+			if hasXValidations(&prop) {
+				return true
+			}
```

Comment:
@liggitt Since this is a case where a prop pointer is borrowed by hasXValidations only for the duration of the call, do you think it's acceptable to do `for _, prop := range s.Properties { hasXvalidations(&prop) ...` in this case (take a pointer of a iteration var)?

Reply:
yeah, when nothing escapes the loop, using the iterator var directly is fine

a non-pointer boolean return clearly can't be retaining the input